homeassistant/components/aussie_broadband/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The Aussie Broadband integration."""
+from __future__ import annotations
+
+from aiohttp import ClientError
+from aussiebb.asyncio import AussieBB, AuthenticationException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_SERVICES, DOMAIN
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Aussie Broadband from a config entry."""
+
+    # Login to the Aussie Broadband API and retrieve the current service list
+    try:
+        client = AussieBB(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            async_get_clientsession(hass),
+        )
+        await client.login()
+        all_services = await client.get_services()
+
+    except AuthenticationException as exc:
+        raise ConfigEntryAuthFailed() from exc
+    except ClientError as exc:
+        raise ConfigEntryNotReady() from exc
+
+    # Filter the service list to those that are enabled in options
+    services = [
+        s for s in all_services if str(s["service_id"]) in entry.options[CONF_SERVICES]
+    ]
+
```

Comment:
What type of services is this and why is a user allowed to control this?

https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md

Reply:
So a service can be a internet connection at a premises, or a mobile phone collection.

Each "service" requires a unique API call to retrieve its data, so instead of assuming the user wants the integration to poll for data from every single service they are given the choice at setup, and using configure.

As it does require extra requests, I believe it meets the requirements of that Architecture decision.

Reply:
That it does, just wanted to make sure  :+1: